nilearn/maskers/tests/test_surface_labels_masker.py

```diff
@@ -0,0 +1,274 @@
+
+
+@pytest.mark.parametrize(
+    "estimator, check, name",
+    check_estimator(
+        estimator=[SurfaceMasker()], extra_valid_checks=extra_valid_checks
+    ),
+)
+def test_check_estimator(estimator, check, name):  # noqa: ARG001
+    """Check compliance with sklearn estimators."""
+    check(estimator)
+
+
+@pytest.mark.xfail(reason="invalid checks should fail")
+@pytest.mark.parametrize(
+    "estimator, check, name",
+    check_estimator(
+        estimator=[SurfaceMasker()],
+        valid=False,
+        extra_valid_checks=extra_valid_checks,
+    ),
+)
+def test_check_estimator_invalid(estimator, check, name):  # noqa: ARG001
+    """Check compliance with sklearn estimators."""
+    check(estimator)
+
+
+def test_surface_label_masker_fit(surf_label_img):
+    """Test fit and check estimated attributes.
+
+    0 value in data is considered as background
+    and should not be listed in the labels.
+    """
+    masker = SurfaceLabelsMasker(labels_img=surf_label_img)
+    masker = masker.fit()
+    assert masker.n_elements_ == 1
+    assert masker._labels_ == [1]
+    assert masker._label_names_ == ["1"]
+    assert masker._reporting_data is not None
+
+
+def test_surface_label_masker_fit_with_names(surf_label_img):
+    """Check passing labels is reflected in attributes.
+
+    - the value corresponding to 0 (background) is omitted
+    - extra value provided (foo) are not listed in attributes
+    """
+    masker = SurfaceLabelsMasker(
+        labels_img=surf_label_img, labels=["background", "bar", "foo"]
+    )
+    masker = masker.fit()
+    assert masker.n_elements_ == 1
+    assert masker._labels_ == [1]
+    assert masker._label_names_ == ["bar"]
+
+
+def test_surface_label_masker_fit_no_report(surf_label_img):
+    """Check no report data is stored."""
+    masker = SurfaceLabelsMasker(labels_img=surf_label_img, reports=False)
+    masker = masker.fit()
+    assert masker._reporting_data is None
+
+
+def test_surface_label_masker_transform(surf_label_img, surf_img):
+    """Test transform extract signals."""
+    masker = SurfaceLabelsMasker(labels_img=surf_label_img)
+    masker = masker.fit()
+
+    # only one 'timepoint'
+    signal = masker.transform(surf_img())
+
+    assert isinstance(signal, np.ndarray)
+    n_labels = len(masker._labels_)
+    assert signal.shape == (1, n_labels)
+
+    # 5 'timepoint'
+    n_timepoints = 5
+    signal = masker.transform(surf_img((n_timepoints,)))
+
+    assert isinstance(signal, np.ndarray)
+    assert signal.shape == (n_timepoints, n_labels)
+
+
+def test_surface_label_masker_transform_check_output(surf_mesh, surf_img, rng):
+    """Check actual content of the transform.
+
+    - Use a label mask with more than one label.
+    - Use data with known content and expected mean.
+      and background label data has random value.
+    - Check that output data is properly averaged,
+      even when labels are spread across hemispheres.
```

Comment:
@man-shu 
can you read this test and tell me if this makes sense to you?
I wanted to have something that does more than just checking the shapes of the output.

Reply:
Yes indeed! Very clear!

Reply:
I just realized that in this test, we weren't actually using the data with 2 timepoints and transforming with the masker.

- Will update in #4770